rust/crd/src/ldap.rs

```diff
@@ -0,0 +1,241 @@
+        lines.insert(
+            "druid.auth.authenticator.Ldap.credentialsValidator.userAttribute".to_string(),
+            Some(self.provider.ldap_field_names.uid.to_string()),
+        );
+        lines.insert(
+            "druid.auth.authenticator.Ldap.credentialsValidator.userSearch".to_string(),
+            Some(self.provider.search_filter.to_string()),
+        );
+        lines.insert(
+            "druid.auth.authenticator.Ldap.authorizerName".to_string(),
+            Some("LdapAuthorizer".to_string()),
+        );
+    }
+
+    fn add_escalator_lines(&self, lines: &mut BTreeMap<String, Option<String>>) {
+        lines.insert(
+            "druid.escalator.type".to_string(),
+            Some("basic".to_string()),
+        );
+        lines.insert(
+            "druid.escalator.internalClientUsername".to_string(),
+            Some("druid_system".to_string()), // TODO: replace with sed-placeholder xxx_druid_system_internal_user_xxx
+        );
+        lines.insert(
+            "druid.escalator.internalClientPassword".to_string(),
+            Some("druid_system_pass".to_string()), // TODO: replace with sed-placeholder
+        );
+        lines.insert(
+            "druid.escalator.authorizerName".to_string(),
+            Some("DruidSystemAuthorizer".to_string()),
+        );
+    }
+
+    fn add_authorizer_lines(&self, lines: &mut BTreeMap<String, Option<String>>) {
+        lines.insert(
+            "druid.auth.authorizers".to_string(),
+            Some(r#"["LdapAuthorizer", "DruidSystemAuthorizer"]"#.to_string()),
+        );
+        lines.insert(
+            "druid.auth.authorizer.DruidSystemAuthorizer.type".to_string(),
+            Some(r#"allowAll"#.to_string()),
+        );
+        lines.insert(
+            "druid.auth.authorizer.LdapAuthorizer.type".to_string(),
+            Some(r#"allowAll"#.to_string()),
+        );
+    }
+
+    pub fn generate_runtime_properties_config_lines(&self) -> BTreeMap<String, Option<String>> {
+        let mut lines: BTreeMap<String, Option<String>> = BTreeMap::new();
+
+        lines.insert(
+            "druid.auth.authenticatorChain".to_string(),
+            Some(r#"["DruidSystemAuthenticator", "Ldap"]"#.to_string()),
+        );
+
+        self.add_druid_system_authenticator_lines(&mut lines);
+        self.add_ldap_authenticator_lines(&mut lines);
+        self.add_escalator_lines(&mut lines);
+        self.add_authorizer_lines(&mut lines);
+
+        lines
+    }
+
+    fn is_ssl_enabled(&self) -> bool {
+        self.provider.tls.is_some()
+    }
+
+    fn get_ldap_protocol_and_port(&self) -> (String, u16) {
+        let protocol = if self.is_ssl_enabled() {
+            "ldaps".to_string()
+        } else {
+            "ldap".to_string()
+        };
+
+        let port = if let Some(port) = self.provider.port {
+            port
+        } else if self.is_ssl_enabled() {
+            DEFAULT_LDAP_TLS_PORT
+        } else {
+            DEFAULT_LDAP_PORT
+        };
```

Comment:
Please use `LdapAuthenticationProvider::default_port` as in https://github.com/stackabletech/superset-operator/blob/fd950b84d8e4e14414aa24eb361c51fdaf2ca319/rust/operator-binary/src/config.rs#L89
DEFAULT_LDAP_TLS_PORT and DEFAULT_LDAP_PORT can be removed

I'm also absolutely happy to keep all of this within `credentials_validator_url()`

Reply:
That particular function outputs different ports than we use here: https://github.com/stackabletech/operator-rs/blob/4dba6f682bb191f5fc807bab50eb0b92d7bb63fb/src/commons/ldap.rs#L34

So we want the behaviour of the code to use well-known ports, but prefer the +1k numbers in our tests. Is that correct?

Reply:
Sound right to me :+1: